eclipse-glsp/glsp-client
# Ensure we apply feedback actions already on SetModel

#### What it does

Ensure that feedback actions are already applied on set-model and not only on update-model.
Fixes https://github.com/eclipse-glsp/glsp/issues/1239

#### How to test

In the workflow example, you can add a startup listener that adds feedback early on, e.g.,
```
import { IDiagramStartup, IFeedbackActionDispatcher, MaybePromise, SelectAction, TYPES } from '@eclipse-glsp/client';
import { inject, injectable } from 'inversify';

@injectable()
export class StartupFeedback implements IDiagramStartup {
    @inject(TYPES.IFeedbackActionDispatcher) protected feedbackDispatcher: IFeedbackActionDispatcher;

    preRequestModel(): MaybePromise<void> {
        this.feedbackDispatcher.registerFeedback(this, [SelectAction.setSelection(['task0'])]);
    }
}
```

With this (and the correct binding in the module) you should see that we have a select feedback registered and applied even if all you do is open the model.

#### Follow-ups

#### Changelog

Method `getFeedbackCommands` was moved from `FeedbackAwareUpdateModelCommand` to `IFeedbackEmitter` for re-use. `FeedbackAwareUpdateModelCommand.getPriority` was replaced by a generic `Prioritized.getPriority`.
